ext/web/02_timers.js

```diff
@@ -38,371 +23,118 @@ function opNow() {
-    const cancelRid = op_timer_handle();
-    timerInfo = { cancelRid, isRef: true, promise: null };
-
-    // Step 4 in "run steps after a timeout".
-    MapPrototypeSet(activeTimers, id, timerInfo);
+function setImmediate(callback, ...args) {
+  if (args.length > 0) {
+    const unboundCallback = callback;
+    callback = () => ReflectApply(unboundCallback, window, args);
   }
 
-  // 3. If the surrounding agent's event loop's currently running task is a
-  // task that was created by this algorithm, then let nesting level be the
-  // task's timer nesting level. Otherwise, let nesting level be zero.
-  // 4. If timeout is less than 0, then set timeout to 0.
-  // 5. If nesting level is greater than 5, and timeout is less than 4, then
-  // set timeout to 4.
-  //
-  // The nesting level of 5 and minimum of 4 ms are spec-mandated magic
-  // constants.
-  if (timeout < 0) timeout = 0;
-  if (timerNestingLevel > 5 && timeout < 4 && respectNesting) timeout = 4;
-
-  // 9. Let task be a task that runs the following steps:
-  const task = {
-    action: () => {
-      // 1. If id does not exist in global's map of active timers, then abort
-      // these steps.
-      //
-      // This is relevant if the timer has been canceled after the sleep op
-      // resolves but before this task runs.
-      if (!MapPrototypeHas(activeTimers, id)) {
-        return;
-      }
-
-      // 2.
-      // 3.
-      if (typeof callback === "function") {
-        try {
-          FunctionPrototypeCall(
-            callback,
-            globalThis,
-            ...new SafeArrayIterator(args),
-          );
-        } catch (error) {
-          reportException(error);
-        }
-      } else {
-        indirectEval(callback);
-      }
-
-      if (repeat) {
-        if (MapPrototypeHas(activeTimers, id)) {
-          // 4. If id does not exist in global's map of active timers, then
-          // abort these steps.
-          // NOTE: If might have been removed via the author code in handler
-          // calling clearTimeout() or clearInterval().
-          // 5. If repeat is true, then perform the timer initialization steps
-          // again, given global, handler, timeout, arguments, true, and id.
-          initializeTimer(callback, timeout, args, true, id);
-        }
-      } else {
-        // 6. Otherwise, remove global's map of active timers[id].
-        core.tryClose(timerInfo.cancelRid);
-        MapPrototypeDelete(activeTimers, id);
-      }
-    },
-
-    // 10. Increment nesting level by one.
-    // 11. Set task's timer nesting level to nesting level.
-    nestingLevel: timerNestingLevel + 1,
-  };
-
-  // 12. Let completionStep be an algorithm step which queues a global task on
-  // the timer task source given global to run task.
-  // 13. Run steps after a timeout given global, "setTimeout/setInterval",
-  // timeout, completionStep, and id.
-  runAfterTimeout(
-    task,
-    timeout,
-    timerInfo,
-    repeat,
+  return core.queueImmediate(
```

Comment:
Is there a way we can unref it? IIRC there's an issue about Node compat `setImmediate` API not having `unref()` method